code/datums/actions/mobs/charge.dm

```diff
@@ -166,7 +166,8 @@
 	living_target.apply_damage(damage_dealt, BRUTE, wound_bonus = CANT_WOUND)
 	playsound(get_turf(living_target), 'sound/effects/meteorimpact.ogg', 100, TRUE)
 	shake_camera(living_target, 4, 3)
-	shake_camera(source, 2, 3)
+	if(isliving(source))
```

Comment:
does this fix like, an issue that doesn't come up often but could in theory?

Reply:
it's literally a mob ability, so source should always be a mob. a QELING check would likely be more appropriate.

Reply:
> does this fix like, an issue that doesn't come up often but could in theory? what is source here?

source is any atom movable according to the arguments of the proc. that's what i predicated my decision based on

```dm
/datum/action/cooldown/mob_cooldown/charge/proc/hit_target(atom/movable/source, atom/target, damage_dealt)
```

Reply:
source is the thing that owns the action button, could likely assert mob and it'd be fine
target is what we end up hitting, but it's typechecked anready